x/bank/keeper/keeper.go

```diff
@@ -355,6 +367,56 @@ func (k BaseKeeper) SendCoinsFromAccountToModule(
 	return k.SendCoins(ctx, senderAddr, recipientAcc.GetAddress(), amt)
 }
 
+
+// LazyDepositFromAccountToModule transfers coins from an AccAddress to a ModuleAccount.
+// It deducts the balance from an accAddress and stores the balance in a mapping for ModuleAccounts.
+// In the EndBlocker, it will then perform one deposit for each module account.
+// It will panic if the module account does not exist.
+func (k BaseKeeper) LazySendCoinsFromAccountToModule(
+	ctx sdk.Context, senderAddr sdk.AccAddress, recipientModule string, amount sdk.Coins,
+) error {
+	// Deducts Fees from the Sender Account
+	err := k.subUnlockedCoins(ctx, senderAddr, amount)
+	if err != nil {
+		return err
+	}
+	k.LazyDepositToModule(recipientModule, amount)
+
+	return nil
+}
+
+func (k BaseKeeper) LazyDepositToModule(recipientModule string, amount sdk.Coins) {
+	k.moduleAccountDepositMappingLock.Lock()
+	defer k.moduleAccountDepositMappingLock.Unlock()
+
+	newAmount := amount
+	if v, ok := k.moduleAccountDepositMapping[recipientModule]; ok {
+		newAmount = v.Add(amount...)
+	}
+	k.moduleAccountDepositMapping[recipientModule] = newAmount
+}
+
+// Iterates on all the lazy deposits and deposit them into the store
+func (k BaseKeeper) WriteLazyDepositsToModuleAccounts(ctx sdk.Context) []abci.Event {
```

Comment:
don't we also need to call this in the bank `endblock` logic?

Reply:
I was thinking of calling it in the sei-chain app - the baseapp in sei-cosmos doesn't have access to the bank keeper